src/ILCompiler.Compiler/src/Compiler/CompilerTypeSystemContext.cs

```diff
@@ -380,6 +380,11 @@ public IEnumerable<string> GetParameterNamesForMethod(MethodDesc method)
             foreach (var parameterHandle in parameters)
             {
                 Parameter p = ecmaMethod.MetadataReader.GetParameter(parameterHandle);
+
+                // Parameter with sequence number 0 refers to the return parameter
+                if (p.SequenceNumber == 0)
+                    continue;
+
```

Comment:
The ECMA spec says that the sequence should be sorted, but it can have holes in it ... so there is a theoretical bug here (cannot think of a compiler that would generate the holes). Maybe add TODO?